test/plugin_functional/test_suites/core_plugins/applications.ts

```diff
@@ -38,6 +41,19 @@ export default function ({ getService, getPageObjects }: PluginFunctionalProvide
       search,
     });
 
+  // Inspired by Dashboard implementations
+  // intention: Use navigateToAppFromAppsMenu rather than appsMenu.clickLink directly,
+  async function navigateToAppFromAppsMenu(title: string) {
+    await retry.try(async () => {
+      await appsMenu.clickLink(title);
+      await waitUntilLoadingIsDone();
+      const currentUrl = await browser.getCurrentUrl();
+      if (!currentUrl.includes('app/foo/home')) {
+        throw new Error(`Not in dashboard application after clicking 'Dashboard' in apps menu`);
```

Comment:
We probably don't need this check, especially given the `navigateToAppFromAppsMenu` accepts any `title`, so (in theory) we're not necessarily going to the `foo` app?